### PR TITLE
UI custom decision texts

### LINKED ISSUE #1240: Cricket:  Sonic Amplifier has vague prompt
**Describe the bug**
The prompt for Sonic Amplifier (after dealing sonic damage) says "do you want to move a card" - it would be clearer to say "Cricket,  do you want to put the top card of your deck beneath Sonic Amplifier?" or something similar.  

**What does it affect**
Card(s): Sonic Amplifier
Deck(s): Cricket

**Screenshots**
image

### LINKED ISSUE #1242: Echelon:  Find a way in prompts Yes or No
**Describe the bug**
Find a way in prompts a vague "Yes or no" question

**What does it affect**
Card(s): Find a way in
Deck(s): Echelon

**Expected behavior**
A more explanatory prompt,  perhaps something like "X,  skip your power phase and play an additional card during your play phase?"

**Screenshots**
image

### LINKED ISSUE #1249: Yes and no
[log.txt](https://github.com/SotMSteamMods/CauldronMods/files/6341843/log.txt)
[output_log.txt](https://github.com/SotMSteamMods/CauldronMods/files/6341844/output_log.txt)

### LINKED ISSUE #1255: Chaos Butterfly's decision text asks the wrong question
**Describe the bug**
When Chaos Butterfly is destroyed, it asks the player "Do you want to move Chaos Butterfly?" rather than something like "Do you want to move 2 Day cards?".

**To Reproduce**
1. Fight the Mistress of Fate with your environment and heroes of choice
2. Destroy Chaos Butterfly with at least 2 Day cards face-up
3. Game asks "Do you want to move Chaos Butterfly?"

**Expected behavior**
The when-destroyed effect says "the players may swap the positions of 2 face-up Day cards." I would expect a decision question along the lines of "do you want to swap 2 Day cards?".

**Screenshots**
[Decision text](https://cdn.discordapp.com/attachments/779089246320984095/836314173960224768/unknown.png)

### PR DESCRIPTION
**Pull request by wrhyme29**
Closes #1249 
Closes #1255 
Closes #1242 
Closes #1240

### CODE REVIEW

File: CauldronMods/Controller/Heroes/Echelon/Cards/SurpriseAttackCardController.cs [hunk 1]
```diff
@@ -130,5 +133,14 @@ private IEnumerator ChangeDamageTypeFromEffectResponse(AddStatusEffectAction se)
                 }
             }
         }
+
+        public override CustomDecisionText GetCustomDecisionText(IDecision decision)
+        {
+
+            return new CustomDecisionText($"Do you want to change all damage dealt from the status effect made by {PowerSourceTitleForCustomText} to Psychic?", $"Should they change all damage dealt from the status effect made by {PowerSourceTitleForCustomText} to Psychic?", $"Vote for if they should change all damage dealt from the status effect made by {PowerSourceTitleForCustomText} to Psychic?", "change damage to Psychic");
```
Comment (cdallasanta) [inline CauldronMods/Controller/Heroes/Echelon/Cards/SurpriseAttackCardController.cs:140]: I'm not sure how to trigger this dialog to test it. I can see it is called on line 35, but can't parse out how to make that happen.

**Review by cdallasanta — APPROVED** [review]
All looks good except for the one text I couldn't test.

Perhaps outside of the scope of this PR: I see for the "vote for if..." lines, we sometimes have them ending in question marks and sometimes nothing. Is the question mark ending the desired one? It appears more frequently.

PR approved. If that one text line is good and the question mark is as intended, go ahead and merge.

**Comment by wrhyme29** [comment]
I'd say it's not intended but also not super important. It only comes up in multiplayer when a player who plays this goes AFK. 

Feel free to change it.

File: CauldronMods/Controller/Heroes/Echelon/Cards/SurpriseAttackCardController.cs (hunk 1 above)
Comment (LNBertolini) [reply CauldronMods/Controller/Heroes/Echelon/Cards/SurpriseAttackCardController.cs:140]: That bit is called when the power sets up a continuing status effect that can cause damage - something like Road Warrior Mainstay or Omnitron-U's innate powers.

**Pull request merged**